### ISSUE TITLE
Random memory corruption with GCC 4.9.2 (and failure to compile the Qt5 demo)

### ISSUE THREAD

**Issue by Syniurge** [Syniurge/Calypso#27]
When compiled with (Ubuntu 15.04) GCC 4.9.2, LDC + Calypso fails to compile the Qt5 demo. It turned out that sometimes a vptr gets corrupted, sometimes it's the last bytes of a Type... and I haven't been able to pinpoint when the corruption happens, because using a breakpoint and querying the values through gdb would always prevent the corruption from happening or wreaking havoc and the compilation of the Qt5 demo would terminate successfully.

None of those random issues occurred with Calypso built with Clang 3.6:

``` bash
    export CC=clang
    export CXX=clang++
    cmake (...)
```

The corruption seems related to the recently enabled instantiation of C++ templated constructors because it's always them which get corrupted. Any idea on how to track the issue down? For now I have given up and switched to Clang, and will add a note to the README if this doesn't get fixed.

**Comment by Syniurge** [comment]
Okay.. with my recent changes not yet committed I still get corruption with Clang builds and "info vtbl" returns the exact same garbage that the GCC build did:

```
    (gdb) info vtbl s
    info vtbl s
    vtable for 'Dsymbol' @ 0xb3b1e20 (subobject @ 0xcf313f0):
    [0]: 0x65c2790 <vtable for clang::ClassTemplateSpecializationDecl+16>
    [1]: 0xb3b24b8
    [2]: 0xb3b1ef2
    [3]: 0xa000e01d7f9e25c6
    [4]: 0xa841c48
    [5]: 0xb3b1fa0
    [6]: 0x7f9d435e
    [7]: 0x21d
    [8]: 0xa5c3e80
    [9]: 0xb3b4030
    [10]: 0xb7d7110
    [11]: 0xb3b1ed3
    [12]: 0x800004b
    [13]: 0x0
    [14]: 0x0
    [15]: 0xb3b1f00
    [16]: 0x0
    (...)
```

Those lines were the same I got with those corrupted templated ctors, so definitely not a GCC problem.

**Comment by Syniurge** [comment]
Fixed with cd6b9bb45c41596ebe49149738f8e9f07142bf8c.
